lib/src/executables/react16_upgrade.dart

```diff
@@ -28,29 +32,134 @@ const _changesRequiredOutput = """
+    if (rec.message != '') {
+      final prefix = '[${rec.level}] ${rec.loggerName}: ';
+      print('$prefix${rec.message}');
+    }
+    if (rec.error != null) {
+      print(rec.error.toString());
+    }
+    if (rec.stackTrace != null) {
+      print(rec.stackTrace.toString());
+    }
+  });
+  react16CodemodLogger
+      .info('Checking if project needs to run $react16CodemodName...');
+  try {
+    pubspecYaml = loadYaml(File('pubspec.yaml').readAsStringSync());
+  } catch (e, stackTrace) {
+    if (e is FileSystemException) {
+      react16CodemodLogger.warning(
+          'Could not find pubspec.yaml; exiting codemod.', e);
+    } else if (e is YamlException) {
+      react16CodemodLogger.warning(
+          'pubspec.yaml is unable to be parsed; exiting codemod.',
+          e,
+          stackTrace);
+    } else {
+      react16CodemodLogger.warning(
+          'pubspec.yaml is unable to be parsed; exiting codemod.',
+          e,
+          stackTrace);
+    }
+  }
+  for (var package in packagesToCheckFor.entries) {
+    var constraint = getDependencyVersion(package.key);
+    if (constraint != null) {
+      foundReactOrOverReact = true;
+      react16CodemodLogger
+          .info('Found ${package.key} with version ${constraint}');
+      // Found it so lets add it to the in transition list to false until its
+      // validated that we know it is in transtition.
+      inTransition[package.key] = false;
+    }
+    if (constraint != null &&
+        !constraint.isAny &&
+        constraint.allowsAny(package.value[0]) &&
+        constraint.allowsAny(package.value[1])) {
+      inTransition[package.key] = true;
+    }
+  }
+  if (foundReactOrOverReact) {
+    if (inTransition.isNotEmpty && inTransition.values.every((val) => val)) {
+      react16CodemodLogger.info('Starting $react16CodemodName...');
+      final query = FileQuery.dir(
+        pathFilter: isDartFile,
+        recursive: true,
+      );
+      exitCode = runInteractiveCodemodSequence(
+        query,
+        [
+          ReactDomRenderMigrator(),
+          ReactStyleMapsUpdater(),
+        ],
+        args: args,
+        defaultYes: true,
+        changesRequiredOutput: _changesRequiredOutput,
+      );
+
+      final logger = Logger('over_react_codemod.fixmes');
+      for (var dartFile in query.generateFilePaths()) {
+        final dartSource = File(dartFile).readAsStringSync();
+        if (dartSource.contains('[ ] $manualValidationCommentSubstring')) {
+          logger.severe(
+              'over_react_codemod validation comments are unaddressed in $dartFile');
+          exitCode = 1;
+        }
+      }
+      react16CodemodLogger.info('Finished Running $react16CodemodName!');
+    } else {
+      react16CodemodLogger.warning(
+          'pubspec.yaml does not have transition versions of react or over_react; exiting codemod.\n' +
+              'if you would like to run the react 16 codemod first run ' +
```

Comment:
#nit capitalization/grammar
```suggestion
              'If you would like to run the React 16 codemod, first run ' +
```